lib/stages/ReadGain.cpp

```diff
@@ -199,34 +195,41 @@ void ReadGain::read_gain_tracking() {
     FILE* ptr_myfile;
     char filename[256];
     bool all_beams_successful_update = true;
-    for (int b = 0; b < _num_beams; b++) {
-        snprintf(filename, sizeof(filename), "%s/quick_gains_%04d_reordered.bin",
-                 _gain_dir_tracking[b].c_str(), freq_idx);
-        INFO("TRACKING Loading gains from {:s}", filename);
-        ptr_myfile = fopen(filename, "rb");
-        if (ptr_myfile == nullptr) {
-            WARN("GPU Cannot open gain file {:s}", filename);
-            all_beams_successful_update = false;
-            for (uint i = 0; i < _num_elements; i++) {
-                out_frame_tracking[(b * _num_elements + i) * 2] = default_gains[0];
-                out_frame_tracking[(b * _num_elements + i) * 2 + 1] = default_gains[1];
-            }
-        } else {
-            if (_num_elements
-                != fread(&out_frame_tracking[b * _num_elements * 2], sizeof(float) * 2,
-                         _num_elements, ptr_myfile)) {
-                WARN("Gain file ({:s}) wasn't long enough! Something went wrong, using default "
-                     "gains",
-                     filename);
+    {
+        std::lock_guard<std::mutex> lock(mux);
```

Comment:
Only lock the check and removal of the of the last beam from the queue.  Otherwise this blocks for as long as it takes the read the gain file (which could be a little while with NFS).